BootloaderCommonPkg/Library/BootloaderCommonLib/BootloaderCommonLib.c

```diff
@@ -359,22 +355,63 @@ GetComponentInfoByPartition (
       //
       if ( ((EntryDesc.Flags & (FLASH_MAP_FLAGS_NON_REDUNDANT_REGION | FLASH_MAP_FLAGS_NON_VOLATILE_REGION)) != 0) ||
            (((IsBackupPartition ? FLASH_MAP_FLAGS_BACKUP : 0) ^ (EntryDesc.Flags & FLASH_MAP_FLAGS_BACKUP)) == 0) ) {
-        PcdBase = (UINT32) (RomBase + EntryDesc.Offset);
-        PcdSize = EntryDesc.Size;
+        Entry = (FLASH_MAP_ENTRY_DESC *)&EntryDesc;

```

Comment:
This is not correct.  
You need to  change the function parameter  to   OUT FLASH_MAP_ENTRY_DESC  **Entry,
and then do   *Entry  = (FLASH_MAP_ENTRY_DESC *)&EntryDesc;

Reply:
I changed implementation, this would be better I guess, please take a look